src/util/contentful/richTextRenderer.js

```diff
@@ -10,50 +10,77 @@ import { BLOCKS, INLINES } from '~/@contentful/rich-text-types';
 import { documentToHtmlString } from '~/@contentful/rich-text-html-renderer';
 
 /**
- * Returns html string to render a contentful asset as a vue component
+ * Returns html string from rich text nodes
  *
- * @param {object} contentfulAssetObject Content of a contentful asset object
+ * @param {object} content Content of a contentful rich text field
  * @returns {string} String of html to render
  */
-const assetRenderer = contentfulAssetObject => {
-	const isVideo = contentfulAssetObject?.file?.contentType.includes('video');
-	const isImage = contentfulAssetObject?.file?.contentType.includes('image');
-	if (isImage) {
-		return `
+export function richTextRenderer(content) {
+	/**
+	 * Returns html string to render a contentful asset as a vue component
+	 *
+	 * @param {object} contentfulAssetObject Content of a contentful asset object
+	 * @returns {string} String of html to render
+	 */
+	const assetRenderer = contentfulAssetObject => {
+		const isVideo = contentfulAssetObject?.file?.contentType.includes('video');
+		const isImage = contentfulAssetObject?.file?.contentType.includes('image');
+		if (isImage) {
+			return `
 			<kv-contentful-img
 				contentful-src="${encodeURI(contentfulAssetObject?.file?.url)}"
 				alt="${contentfulAssetObject?.description}"
 				height="${contentfulAssetObject?.file?.details?.image?.height}"
 				width="${contentfulAssetObject?.file?.details?.image?.width}"/>
 			`;
-	} if (isVideo) {
+		} if (isVideo) {
 		// video media
-		return `
+			return `
 			<video
 				:src="${encodeURI(contentfulAssetObject?.file?.url)}"
 				autoplay
 				loop
 				muted
 				playsinline
 			></video>`;
-	}
-	return '';
-};
+		}
+		return '';
+	};
+
+	/**
+	 * Returns html string to render a contentful entry, possibly as a vue component
+	 *
+	 * @param {object} contentfulEntryNode Contentful rich text node for an entry
+	 * @returns {string} String of html to render
+	 */
+	const entryRenderer = contentfulEntryNode => {
+		const isRichTextContent = contentfulEntryNode?.data?.target?.sys?.contentType?.sys?.id === 'richTextContent';
+		const isButton = contentfulEntryNode?.data?.target?.sys?.contentType?.sys?.id === 'button';
+		if (isRichTextContent) {
+			return `
+				<div class="tw-prose">${richTextRenderer(contentfulEntryNode?.data?.target?.fields?.richText)}</div>
+			`;
+		}
+		if (isButton) {
+			const analytics = contentfulEntryNode?.data?.target?.fields?.analyticsClickEvent ?? [];
+			return `
+				<kv-button
+					variant="${contentfulEntryNode?.data?.target?.fields?.style ?? 'primary'}"
+					href="${contentfulEntryNode?.data?.target?.fields?.webLink ?? '#'}"
+					v-kv-track-event="['${analytics.join("','")}']"
```

Comment:
This is not working, I'm looking into it

Reply:
Interesting. Do you get the correct markup here?

Reply:
It is working, made some slight modifications so we arent firing the kv-track-event directive when the field does not exist in contentful